packages/aws-cdk/lib/api/cxapp/exec.ts

```diff
@@ -75,14 +77,40 @@ export async function execProgram(aws: SdkProvider, config: Configuration): Prom
   env[cxapi.OUTDIR_ENV] = outdir;
 
   // Send version information
-  env[cxapi.CLI_ASM_VERSION_ENV] = cxapi.CLOUD_ASSEMBLY_VERSION;
   env[cxapi.CLI_VERSION_ENV] = versionNumber();
 
   debug('env:', env);
 
   await exec();
 
-  return new cxapi.CloudAssembly(outdir);
+  return createAssembly(outdir);
+
+  function createAssembly(directory: string) {
+
+    const assembly = new cxapi.CloudAssembly(directory);
+    assertAssemblyVersion(assembly);
+    return assembly;
+
+  }
+
+  function assertAssemblyVersion(cloudAssembly: cxapi.CloudAssembly) {
```

Comment:
I feel like this should be a utility of the cloud-assembly-schema module, no? I.e. answer the question "is this cloud assembly (directory) supported by this module?" Even better, should just be part of `loadAssembly`

Reply:
I like it 👍